build/pool-manager/entrypoint.sh

```diff
@@ -1,8 +1,34 @@
-#!/bin/sh
+#!/bin/bash
 
 set -ex
 
+#sigterm caught SIGTERM signal and forward it to pool_manager process
+_sigterm() {
+  echo "[entrypoint.sh] caught SIGTERM signal forwarding to pid [$pool_manager_pid]."
+  kill -TERM "$pool_manager_pid" 2> /dev/null
+  waitForChildProcessToFinish
+}
+
+#sigint caught SIGINT signal and forward it to pool_manager process
+_sigint() {
+  echo "[entrypoint.sh] caught SIGINT signal forwarding to pid [$pool_manager_pid]."
+  kill -INT "$pool_manager_pid" 2> /dev/null
+  waitForChildProcessToFinish
+}
+
+#waitForChildProcessToFinish waits for pool_manager process to finish
+waitForChildProcessToFinish(){
+    while ps -p "$pool_manager_pid" > /dev/null; do sleep 1; done;
+}
+
 rm /usr/local/bin/zrepl
-exec /usr/local/bin/pool-manager start
-exec service ssh start
-exec service rsyslog start
+exec /usr/local/bin/pool-manager start &
```

Comment:
Why do we need `exec` here? Can't we directly start the pool-manager.

question: Does using exec here replaces the current process with the pool manager, or is a fork created automatically?

Reply:
exec replaces the current process with the pool manager.

> Why do we need exec here? Can't we directly start the pool-manager.

Just following whatever exists. Let me try out without exec and observe the behavior.

Reply:
> Can't we directly start the pool-manager.

I will update it to run directly.